### PR TITLE
tests,github,spread: fix debian prepare logic

### PR DESCRIPTION
**Pull request by zyga**
We've been experiencing failures to prepare Debian systems caused by *.pyc files related to one of the log analyzers:

```
dpkg-source: error: cannot represent change to tests/lib/external/snapd-testing-tools/utils/__pycache__/log_helper.cpython-310.pyc: binary file contents changed
```

Since the files may be created at any time, and spread may be used interactively on a stateful host, use defense-in-depth approach by filtering out `*.pyc` files in both spread and tar, and by attempting not to write them when starting log analyzers from the host running spread.

A pair of drive-by `retry` simplifications is included.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -601,6 +601,7 @@ exclude:
     - cmd/autom4te.cache
     - "*.o"
     - "*.a"
+    - "*.pyc"
```
Comment (bboozzoo) [inline spread.yaml:604]: hm likely this is the only change we need

Comment (zyga) [reply spread.yaml:604]: When running interactively we probably also want the one to tar. That is technically the real fix.

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by ernestl — APPROVED** [review]
Thanks

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -870,7 +870,7 @@ jobs:
           # propagated; and we use a subshell as this option could trigger
           # undesired changes elsewhere
           echo "Running command: $SPREAD $RUN_TESTS"
-          (set -o pipefail; $SPREAD -no-debug-output -logs .logs $RUN_TESTS | ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | tee spread.log)
+          (set -o pipefail; $SPREAD -no-debug-output -logs .logs $RUN_TESTS | PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | tee spread.log)
```
Comment (sergiocazzolato) [inline .github/workflows/test.yaml:873]: Perhaps we could just save this env var in a previous step as we do in "name: Prepare test results env and vars"

echo "PYTHONDONTWRITEBYTECODE=1" >> $GITHUB_ENV

Comment (zyga) [reply .github/workflows/test.yaml:873]: I wanted to be precise here but I think we could actually drop this patch entirely, as we are also protected by the changes in spread.yaml and in the prepare script.

**Review by sergiocazzolato — APPROVED** [review]
thanks

**Pull request merged**